awx/ui_next/src/screens/Host/Host.jsx

```diff
@@ -12,217 +19,190 @@ import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
 import HostFacts from './HostFacts';
 import HostDetail from './HostDetail';
-
 import HostEdit from './HostEdit';
 import HostGroups from './HostGroups';
 import HostCompletedJobs from './HostCompletedJobs';
 import { HostsAPI } from '@api';
 
-class Host extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      host: null,
-      hasContentLoading: true,
-      contentError: null,
-      isInitialized: false,
-    };
-    this.loadHost = this.loadHost.bind(this);
-  }
-
-  async componentDidMount() {
-    await this.loadHost();
-    this.setState({ isInitialized: true });
-  }
-
-  async componentDidUpdate(prevProps) {
-    const { location, match } = this.props;
-    const url = `/hosts/${match.params.id}/`;
-
-    if (
-      prevProps.location.pathname.startsWith(url) &&
-      prevProps.location !== location &&
-      location.pathname === `${url}details`
-    ) {
-      await this.loadHost();
-    }
-  }
-
-  async loadHost() {
-    const { match, setBreadcrumb, history, inventory } = this.props;
-
-    this.setState({ contentError: null, hasContentLoading: true });
+function Host({ inventory, i18n, setBreadcrumb }) {
+  const [host, setHost] = useState(null);
+  const [contentError, setContentError] = useState(null);
+  const [hasContentLoading, setHasContentLoading] = useState(true);
+
+  const location = useLocation();
+  const hostsMatch = useRouteMatch('/hosts/:id');
+  const inventoriesMatch = useRouteMatch(
+    '/inventories/inventory/:id/hosts/:hostId'
+  );
+  const baseUrl = hostsMatch ? hostsMatch.url : inventoriesMatch.url;
+  const hostListUrl = hostsMatch
+    ? '/hosts'
+    : `/inventories/inventory/${inventoriesMatch.params.id}/hosts`;
+
+  const loadHost = async () => {
+    setContentError(null);
+    setHasContentLoading(true);
     try {
-      const { data } = await HostsAPI.readDetail(
-        match.params.hostId || match.params.id
-      );
-      this.setState({ host: data });
+      const hostId = hostsMatch
+        ? hostsMatch.params.id
+        : inventoriesMatch.params.hostId;
+      const { data } = await HostsAPI.readDetail(hostId);
+      setHost(data);
 
-      if (history.location.pathname.startsWith('/hosts')) {
+      if (hostsMatch) {
         setBreadcrumb(data);
-      } else {
+      } else if (inventoriesMatch) {
         setBreadcrumb(inventory, data);
       }
-    } catch (err) {
-      this.setState({ contentError: err });
+    } catch (error) {
+      setContentError(error);
     } finally {
-      this.setState({ hasContentLoading: false });
+      setHasContentLoading(false);
     }
+  };
+
+  useEffect(() => {
+    loadHost();
+  }, [location]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
We're not reusing / reloading the hosts in more than one place. Do we really need to move it outside of the `useEffect` block?